### ISSUE TITLE
About the configuration for Base+R in Table 6

### ISSUE THREAD

**Issue by aaronjie** [shariqfarooq123/AdaBins#6]
Hi, there. Thank you for your paper and code. 
I tried to re-train your model named Base+R in Table 6.
As mentioned in your paper, I set Cd=1 which refers to num_classes=1 in https://github.com/shariqfarooq123/AdaBins/blob/2fb686a66a304f0a719bc53d77412460af97fd61/models/unet_adaptive_bins.py#L26. Due to the output should be greater than zero (log used in loss function), I added an extra nn.Sigmoid to force the output to be [0, 1] and scaled it to [0, 10] by multiply 10. These two changes were all made in DecoderBN.
The final model looks like:
```python
class BaseR(nn.Module):  
    def __init__(self):  
        super(BaseR, self).__init__()  
        self.encoder = Encoder()  
        self.decoder = DecoderBN(num_classes=1)  
    def forward(self, x):  
        unet_out = self.decoder(self.encoder(x))  
        return unet_out  
    ....
```
Besides, the loss only contains the Pixel-wise depth loss. The batch size is 10 for gpu memory limitation. All other configurations were the same as yours.
After trained with 25 epochs, I got the loss around 0.4 and the loss curve as:
![image](https://user-images.githubusercontent.com/4651316/103473333-4faada00-4dd2-11eb-8995-4ff9e60e28fb.png)

And the evaluation results are:
Metrics: {'a1': 0.0, 'a2': 0.0, 'a3': 0.0, 'abs_rel': 0.974, 'rmse': 2.795, 'log_10': 1.583, 'rmse_log': 3.649, 'silog': 12.393, 'sq_rel': 2.555}

I don't know what is wrong here. Do you remember your final loss value ? or can you provide your model code for Base+R.

Thank you very much! :)

**Comment by shariqfarooq123** [comment]
Sigmoid is not a good choice for predicting continuous depth values as it saturates strongly towards extremes. 
You can instead use ReLU as the final activation and add an epsilon-shift for SILog loss calculation like so: 
```python
eps = 0.1
loss = silog(input + eps, target + eps, mask=target>0)
```
For your reference, loss curves for the final model ("AdaBins (ours)" in the paper) look like this: 
![image](https://user-images.githubusercontent.com/22071273/103880538-5004e700-50ea-11eb-801c-48cadf6c0303.png)

**Comment by freshjh** [comment]
I also troubled by the training of baseline method. I trained the (Encoder+DecoderBN) with SILog loss while get a unmeaning result  like @aaronjie. Could you please offer the code of baseline method? Look forward to your reply. Thanks! 
![image](https://user-images.githubusercontent.com/70044642/129517500-59bb98f9-b17e-44ee-994b-ff8b027e0029.png)

**Comment by ksh11023** [comment]
Hello,

I have encountered same problem.
Have you solve the problem for Metrics a1,  a2, a3 having value=0? 

Thank you.

**Comment by ksh11023** [comment]
@freshjh 

Hello,

I have encountered the same problem.
Have you solved the problem for metrics a1, a2, a3 having value=0?

Thankyou.
